contracts/oracle/src/lib.rs

```diff
@@ -150,17 +150,25 @@ impl Contract {
         };
 
         self.used_identities.insert(&external_id);
-        ext_registry::ext(self.registry.clone())
+        let result = ext_registry::ext(self.registry.clone())
             .with_attached_deposit(MINT_COST_REG)
             .with_static_gas(Gas::ONE_TERA * 6)
-            .sbt_mint(vec![(claim.claimer, vec![metadata])]);
-
-        // TODO: add callback to undo identity insert if the minting didn't work because of a duplicate mint.
+            .sbt_mint(vec![(claim.claimer, vec![metadata])]).then(Self::ext(env::current_account_id()).sbt_mint_callback(&external_id));
         if let Some(memo) = memo {
             env::log_str(&format!("SBT mint memo: {}", memo));
         }
```

Comment:
let's move this `if` before `let result = ...`

Reply:
DONE